### PR TITLE
add scroolbars to wide graphs.

### PR DESCRIPTION
**Pull request by kjellkvinge**
Ref #599 

This will use all available space to display the graph.

### CODE REVIEW

**Comment by lunny** [comment]
less file?

**Comment by kjellkvinge** [comment]
What do you mean? I just made som changes in the existing template, css and js for the graph.

**Comment by lunny** [comment]
Maybe you have to modify less file since you have modified css file.

**Comment by kjellkvinge** [comment]
There is no less file. Its just a standard css file for this feature.

File: public/css/gitgraph.css [hunk 1]
```diff
@@ -10,6 +10,6 @@ em {font-style:normal;}
 #git-graph-container li a {color:#000000;}
 #git-graph-container li a:hover {text-decoration:underline;}
 #git-graph-container li a em {color:#BB0000;border-bottom:1px dotted #BBBBBB;text-decoration:none;font-style:normal;}
-#rev-container {width:80%}
-#rev-list {margin:0;padding:0 5px 0 0;width:80%}
+#rev-container {width:100%}
+#rev-list {margin:0;padding:0 5px 0 0;width:100%}
```
Comment (bkcsoft) [inline public/css/gitgraph.css]: wouldn't `padding: 0 5px 0 5px; min-width: 95%` be better?

File: templates/repo/graph.tmpl [hunk 2]
```diff
@@ -1,7 +1,7 @@
 {{template "base/head" .}}
 <div class="repository commits">
 	{{template "repo/header" .}}
-	<div class="ui container">
+	<div class="ui">
```
Comment (bkcsoft) [inline templates/repo/graph.tmpl]: for something else (don't remember what ) I did a `container.full` which sets the width to "100%"

File: public/css/gitgraph.css (hunk 1 above)
Comment (kjellkvinge) [reply public/css/gitgraph.css]: did you try this locally. On my test, with the repo from the issue, it looks better using 100%: 

screenshot from 2017-01-09 21-54-28


95%:

screenshot from 2017-01-09 21-54-22

(this was on chrome, around 900px wide)

what do you think?

File: templates/repo/graph.tmpl (hunk 2 above)
Comment (kjellkvinge) [reply templates/repo/graph.tmpl]: thats probably better :)

File: public/css/gitgraph.css (hunk 1 above)
Comment (kjellkvinge) [reply public/css/gitgraph.css]: sorry. did not notice the 'min' part. min-width looks better :)

**Comment by kjellkvinge** [comment]
for further improvements, it might be possible to make the graph more compact, but then We would have to change the lib graph.js, this should probably be done upstream for the lib.

It accepts one parameter for "unitSize", We now use default value, which is 20. But If we could set unitSize-x to 10 and unitSize-y to 20, the graph could be more compact and still have the same height. (the y value have to match the font-size used for the text.)

but, I am not sure how many parallel branches there is useful to display. Maybe there should be a limit?

**Comment by Bwko** [comment]
@kjellkvinge Maybe you could add a horizontal scrollbar. Otherwise we'll keep running into the same problem.

**Comment by Bwko** [comment]
LGTM

**Comment by Bwko** [comment]
@kjellkvinge Nitpicking: could reset the width to 80%? I personally think the width of 100% looks kinda ugly :sweat_smile:

**Comment by kjellkvinge** [comment]
@Bwko yes, this is difficult :)

you mean somethink like this:
```patch
-#git-graph-container {overflow-x:auto; width:100%}
+#git-graph-container {overflow-x:auto; width:80%; margin-left:10%}
```

This will bring more air on the sides, but it looks miss placed when you have a repo with only one, or a few branches in parallell.

it also depends on the screen size.

maybe we should keep the "ui container" unless there is more than 5 or 10 paralell branches??  any thoughts?

**Comment by Bwko** [comment]
@kjellkvinge Thanks :+1: I agree it's hard to get it right. I've tried this solution but it still looks kinda weird. Maybe we should keep the width to 100% width and remove the `.full` class. Perhaps not the best solution but it prevents jumping of the page width.

**Comment by kjellkvinge** [comment]
ok, now trying a new approach. This was originally a PR for using full page width. Now it just adds scrollbars if the graph is to wide. This might solve both maintaining the same layout as the rest of the pages, and still usable even if there is a lot of parallel branches in the graph.

**Comment by Bwko** [comment]
@kjellkvinge I find this to be the best solution thus far. 
Does someone else have an opinion on this?

**Comment by kjellkvinge** [comment]
Not to much opinions out there :) Maybe I can provide some screenshots, or a live demo somewhere, so people can test without without rebuilding?

**Comment by Bwko** [comment]
@kjellkvinge Screenshots are good enough (I guess)

**Comment by lunny** [comment]
LGTM

**Pull request merged**